apps/labs/posts/intel-oneapi-msvc-spin-scipy.md

```diff
@@ -0,0 +1,88 @@
+category: [Developer workflows]
+featuredImage:
+  src: /posts/intel-oneapi-msvc-spin-scipy/scipy_logo_img_featured.png
+  alt: 'SciPy logo'
+hero:
+  imageSrc: /posts/intel-oneapi-msvc-spin-scipy/scipy_logo_img_hero.png
+  imageAlt: 'SciPy logo'
+---
+
+In this blog post, we will be sharing the work we did to enhance developer experience at SciPy. We worked on two major things,
+
+1. Intel oneAPI and MSVC Support in SciPy
+2. Moving away from `dev.py` and migrating to `spin`.
+
+First, let me share the motivation behind these.
+
+Intel offers a complete toolkit for high performance computing (HPC), numerical computing use cases. It goes by the name of [oneAPI](https://www.intel.com/content/www/us/en/developer/tools/oneapi/overview.html). It provides, ICX (for C), ICPX (for C++), IFX (for Fortran) and Math Kernal Library (MKL for optimized math routines like BLAS, LAPACK, fast FFT, etc.) [[1]](https://en.wikipedia.org/wiki/Math_Kernel_Library). Since, SciPy caters to HPC and numerical computing use cases, so supporting building SciPy with Intel oneAPI is a natural requirement. In fact, users reported segmentation fault in `arpack` with `ifx` [[gh-20728]](https://github.com/scipy/scipy/issues/20728) and `scipy.linalg.solve` crash when used with Intel oneAPI on Windows 11 [[gh-21997]](https://github.com/scipy/scipy/issues/21997) are two such examples. Therefore we decided to work on this to enhance the experience of SciPy users who use Intel oneAPI as their toolkit. Regarding MSVC, since it is the de-facto compiler for C/C++ on Windows. So adding support for it is again a natural requirement and also a must.
+
+Now let's see why we decided to spin from `dev.py` to `spin` (pun inteded). `dev.py` (or `do.py` [[2]](https://labs.quansight.org/blog/the-evolution-of-the-scipy-developer-cli)) was added to improve the experience for SciPy developers. Basically, it offers better documentation for the various commands needed (like, `build`, `test`, `shell`) while working on SciPy codebase. It has better looking CLI outputs. However, with time several issues with `dev.py` were reported. For example, in some cases, doit based dev interface interferes with pdb command history [[gh-16452]](https://github.com/scipy/scipy/issues/16452), `dev.py` giving problems in a Windows CI environment because of emojis usage in dev.py [[gh-18046]](https://github.com/scipy/scipy/issues/18046), and dev.py unable to resolve homebrew installed Python correctly [[gh-18998]](https://github.com/scipy/scipy/issues/18998). `spin` easily fixes this issues. [[3]](https://github.com/scipy/scipy/pull/21674), especially those issues arising due to `doit`. `dev.py` depends on `doit` but `spin` doesn't. Also, since `spin` is used by many other projects, so all the improvements in `spin` due to the issues reported by their maintainers will directly benefit SciPy.
```

Comment:
I'd say the long-term maintenance load sharing and the familiarity for contributors coming from `numpy` or other project are the main motivations. The paper cuts & bugs are very minor.

Reply:
Also, it'd be good to say somehting like "`dev.py` was a major improvement in development workflow, and came before `spin`. `spin` took its inspiration [in large part](https://github.com/scientific-python/spin/?tab=readme-ov-file#history) from `dev.py`, and generalized its interface to work with more projects without customization as heavy as `dev.py` has for SciPy".